tests/conftest.py

```diff
@@ -47,13 +59,257 @@ def invalid_cephfs_storageclass(request):
     storageclass = OCS(**yaml_data)
     sc_data = storageclass.create()
 
-    logger.debug('Check that storageclass has assigned creationTimestamp')
+    log.debug('Check that storageclass has assigned creationTimestamp')
     assert sc_data['metadata']['creationTimestamp']
 
     yield sc_data
 
-    logger.info(
+    log.info(
         f"TEARDOWN - removing storageclass "
         f"{request.param['storageclass_name']}"
     )
     storageclass.delete()
+
+
+def cluster_teardown():
+    log.info("Destroying the test cluster")
+    destroy_cluster(ENV_DATA['cluster_path'])
+    log.info("Destroying the test cluster complete")
+
+
+@pytest.fixture(scope="session", autouse=True)
+def cluster(request):
+    log.info("Running OCS basic installation")
+    cluster_path = ENV_DATA['cluster_path']
+    # Add a finalizer to teardown the cluster after test execution is finished
+    if config.teardown:
+        request.addfinalizer(cluster_teardown)
+        log.info("Will teardown cluster because --teardown was provided")
+    # Test cluster access and if exist just skip the deployment.
+    if is_cluster_running(cluster_path):
+        log.info("The installation is skipped because the cluster is running")
+        return
+    elif config.teardown and not config.deploy:
+        log.info("Attempting teardown of non-accessible cluster: %s", cluster_path)
+        return
+    elif not config.deploy and not config.teardown:
+        msg = "The given cluster can not be connected to: {}. ".format(cluster_path)
+        msg += "Provide a valid --cluster-path or use --deploy to deploy a new cluster"
+        pytest.fail(msg)
+    else:
+        log.info("A testing cluster will be deployed and cluster information stored at: %s", cluster_path)
```

Comment:
Wouldn't this deploy new cluster in case of communication problems between `oc` and the already existing cluster, when `--deploy` is specified?

Reply:
Yes, if you have ``--deploy`` specified it will deploy a cluster if the one given can not be connected to.